File: WebRouterApp/Features/Subscribers/Application/Models/SubscriberRecord.cs
using System;
using WebRouterApp.Core.Application.Models;

namespace WebRouterApp.Features.Subscribers.Application.Models
{
    public class SubscriberRecord : IEntity
    {
        public SubscriberRecord(
            string apiKey, 
            string apiSecret, 
            string name,
            string description,
            CoeffKinds coeffKind, 
            double multiplier, 
            TradeKinds tradeKind)
        {
            ApiKey = apiKey;
            ApiSecret = apiSecret;
            Name = name;
            Description = description;
            CoeffKind = coeffKind;
            Multiplier = multiplier;
            TradeKind = tradeKind;
        }

        public Guid Id { get; set; }
        
        public string ApiKey { get; set; }
        public string ApiSecret { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public CoeffKinds CoeffKind { get; set; }
        public double Multiplier { get; set; }
        public TradeKinds TradeKind { get; set; }
    }

    public enum CoeffKinds
    {
        CoeffToSize = 0,
    }

    public enum TradeKinds
    {
        TradeAsMarket = 0,
    }
}

File: WebRouterApp/ClientApp/src/app/subscribers/models/subscriber-record.ts
import { Guid } from '../../shared/types/guid';

export interface ISubscriberRecord {
  id: Guid;
  apiKey: string;
  apiSecret: string;
  name: string;
  description: string;
  multiplier: number;
  coeffKind: CoeffKinds;
  tradeKind: TradeKinds;
}

export enum TradeKinds {
  TradeAsMarket = 0,
}

export namespace TradeKinds {
  export function stringify(tradeKind: TradeKinds): string {
    switch (tradeKind) {
      case TradeKinds.TradeAsMarket:
        return 'Trade as market';
      default:
        return '--';
    }
  }
}

export enum CoeffKinds {
  CoeffToSize = 0,
}

export namespace CoeffKinds {
  export function stringify(coeffKind: CoeffKinds): string {
    switch (coeffKind) {
      case CoeffKinds.CoeffToSize:
        return 'Coefficient to size';
      default:
        return '--';
    }
  }
}
